boring-sys/patches/boring-pq.patch

```diff
@@ -154,3178 +172,3184 @@ index 4c0048e1d..f6ea5c40f 100644
--  // because we require |uint8_t| to be a character type.
--  const uint8_t *state_bytes = (const uint8_t *)ctx->state;
--  while (out_len) {
--    size_t remaining = ctx->rate_bytes - ctx->offset;
--    size_t todo = out_len;
--    if (todo > remaining) {
--      todo = remaining;
--    }
--    OPENSSL_memcpy(out, &state_bytes[ctx->offset], todo);
--    out += todo;
--    out_len -= todo;
--    ctx->offset += todo;
--    if (ctx->offset == ctx->rate_bytes) {
--      keccak_f(ctx->state);
--      ctx->offset = 0;
--    }
--  }
--}
-diff --git a/src/crypto/kyber/kyber.c b/src/crypto/kyber/kyber.c
-index 776c085f9..346d4daec 100644
---- a/src/crypto/kyber/kyber.c
-+++ b/src/crypto/kyber/kyber.c
-@@ -1,833 +1,2252 @@
--/* Copyright (c) 2023, Google Inc.
-- *
-- * Permission to use, copy, modify, and/or distribute this software for any
-- * purpose with or without fee is hereby granted, provided that the above
-- * copyright notice and this permission notice appear in all copies.
-- *
-- * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
-- * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
-- * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
-- * SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
-- * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
-- * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
-- * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE. */
-+// Taken from round 3 public domain reference implementation
-+// 
-+//  https://github.com/pq-crystals/kyber
-+//  8e00ec73035147d18b27d06048dff322f8de1f29
-+//
-+//  with some small modifications:
-+//
-+//   - Merged into one file.
-+//   - Removed 90s version.
-+//   - Seeds are passed as paramters.
-+//   - Changed the API to be more BoringSSL-like
-+//
-+//  TODO
-+//
-+//   - Optimizations
-+//
-+//     The majority of Kyber's time is spent in keccak: generating the matrix
-+//     A, hashing the public key, et cetera. This can be sped up dramatically
-+//     by using a multiway keccak implementation such as f1600x4 on AVX2.
-+//
-+//     Also the NTT and other operations can be sped up with SIMD. This is
-+//     more complex and the gains are more modest. See the avx2 reference
-+//     implementation or https://github.com/cloudflare/circl/tree/main/pke/kyber
-+//
-+//   - Option to keep A stored in private key.
+diff --git a/src/crypto/kyber/fips202.c b/src/crypto/kyber/fips202.c
+new file mode 100644
+index 000000000..9713a4f7e
+--- /dev/null
++++ b/src/crypto/kyber/fips202.c
+@@ -0,0 +1,699 @@
++/* Based on the public domain implementation in crypto_hash/keccakc512/simple/ from
++ * http://bench.cr.yp.to/supercop.html by Ronny Van Keer and the public domain "TweetFips202"
++ * implementation from https://twitter.com/tweetfips202 by Gilles Van Assche, Daniel J. Bernstein,
++ * and Peter Schwabe */
```

Comment:
Does this file need a copyright notice? Likewise for the header and ipdwing.c.

Reply:
Implied by LICENSE-MIT file two folders up.